lib/matplotlib/tests/test_colorbar.py

```diff
@@ -235,6 +235,23 @@ def test_remove_from_figure(use_gridspec):
     assert (pre_position.get_points() == post_position.get_points()).all()
 
 
+def test_remove_from_figure_cl():
+    """
+    Test `remove` with constrained_layout
+    """
+    fig, ax = plt.subplots(constrained_layout=True)
+    sc = ax.scatter([1, 2], [3, 4], cmap="spring")
+    sc.set_array(np.array([5, 6]))
+    fig.draw_without_rendering()
+    pre_position = ax.get_position()
+    cb = fig.colorbar(sc)
+    cb.remove()
```

Comment:
I'd also check that the ax position here was changed.

Reply:
Thats what the rest of the check does?   I'm not following.